File: dependency-mapper/dependency-mapper/src/main/java/com/example/mapper/adapter/TelemetryAdapter.java
package com.example.mapper.adapter;

import java.util.List;

public class TelemetryAdapter implements DataSourceAdapter<List<String>> {
    private final List<String> metrics;

    public TelemetryAdapter(List<String> metrics) {
        this.metrics = metrics;
    }

    @Override
    public List<String> fetch() {
        return metrics;
    }
}

File: service/adapter/TelemetryAdapter.js
const DataSourceAdapter = require('./DataSourceAdapter');

class TelemetryAdapter extends DataSourceAdapter {
  constructor(options = {}) {
    super();
    this.metrics = options.metrics || [];
  }

  fetch() {
    // Return provided metrics; in real world this would query telemetry system
    return this.metrics;
  }
}

module.exports = TelemetryAdapter;
